mangaki/mangaki/wrappers/anilist.py

```diff
@@ -192,335 +200,314 @@ def anilist_id(self) -> int:
+            self.anilist_id,
+            self.title,
+            self.status.value
+        )
 
-        self._auth = r.json()
-        self._session.headers['Authorization'] = 'Bearer ' + self._auth['access_token']
 
-        return self._auth
+class AniList:
+    BASE_URL = 'https://graphql.anilist.co'
 
-    def _is_authenticated(self):
-        if not self.is_available:
-            return False
-        if self._auth is None:
-            return False
-        return self._auth['expires'] > time.time()
+    def __init__(self):
+        self._cache = {}
 
     def _request(self,
-                 datapage: str,
-                 params: Optional[Dict[str, str]] = None,
-                 query_params: Optional[Dict[str, str]] = None):
+                 query: str,
+                 variables: Dict[str, Any]) -> Dict[str, Any]:
         """
-        Request an Anilist API's page (see https://anilist-api.readthedocs.io/en/latest/ for more)
-        >>> self._request("{series_type}/{id}", params={"series_type": "anime", "id": "5"})
-        Returns a series model as a JSON.
-        >>> self._request("browse/{series_type}", params={"series_type": "anime"}, query_params={"year": "2017"})
-        Returns up to 40 small series models where year is 2017 as a JSON.
+        Make a request to AniList's v2 API.
+        :param self: an AniList client
+        :param query: the GraphQL query string
+        :param variables: the GraphQL variables provided for the query
+        :type self: AniList
+        :type query: str
+        :type variables: Dict[str, Any]
+        :return: the data returned for this query
+        :rtype: Dict[str, Any]
+        :raises: an AniListException in case of error
         """
-        if not self._is_authenticated():
-            self._authenticate()
-
-        if params is None:
-            params = {}
 
-        if query_params is None:
-            query_params = {}
-
-        r = self._session.get(urljoin(self.BASE_URL, datapage.format(**params)), params=query_params)
+        r = requests.post(
+            self.BASE_URL,
+            json={
+                'query': query,
+                'variables': variables
+            },
+            headers={
+                'Content-Type': 'application/json',
+                'Accept': 'application/json',
+            }
+        )
         data = r.json()
 
-        if not isinstance(data, list) and data.get('error'):
-            if is_no_results(data['error']):
-                return None
-            elif is_route_not_found(data['error']):
-                raise AniListException('"{}" API route does not exist'.format(datapage))
-            elif is_token_error(data['error']):
-                raise AniListException('token no longer valid or not found')
-            else:
-                raise AniListException(data['error'])
+        if data.get('errors'):
+            for error in data['errors']:
+                raise AniListException(error)
```

Comment:
That means that only one error is raised. Others are silenced.
Maybe let's put an array of errors. It's up to you.